conformance/utils/kubernetes/helpers.go

```diff
@@ -0,0 +1,251 @@
+				t.Errorf("Error listing Pods: %v", err)
+			}
+			for _, pod := range podList.Items {
+				if !findPodConditionInList(t, pod.Status.Conditions, "Ready", "True") {
+					t.Logf("%s/%s Pod not ready yet", ns, pod.Name)
+					return false, nil
+				}
+			}
+		}
+		t.Logf("Gateways and Pods in %s namespaces ready", strings.Join(namespaces, ", "))
+		return true, nil
+	})
+	require.NoErrorf(t, waitErr, "error waiting for %s namespaces to be ready", strings.Join(namespaces, ", "))
+}
+
+// GatewayAndHTTPRoutesMustBeReady waits until the specified Gateway has an IP
+// address assigned to it and the Route has a ParentRef referring to the
+// Gateway. The test will fail if these conditions are not met before the
+// timeouts.
+func GatewayAndHTTPRoutesMustBeReady(t *testing.T, c client.Client, controllerName string, gwNN types.NamespacedName, routeNNs ...types.NamespacedName) string {
+	t.Helper()
+
+	gwAddr, err := WaitForGatewayAddress(t, c, gwNN, 180)
+	require.NoErrorf(t, err, "timed out waiting for Gateway address to be assigned")
+
+	for _, routeNN := range routeNNs {
+		parents := []v1alpha2.RouteParentStatus{{
+			ParentRef: v1alpha2.ParentRef{
+				Name: v1alpha2.ObjectName(gwNN.Name),
+				// TODO: Does this need to be specified in status for same
+				// namespace resources?
+				// Namespace: &ns,
+			},
+			ControllerName: v1alpha2.GatewayController(controllerName),
+			Conditions: []metav1.Condition{{
+				Type:   string(v1alpha2.ConditionRouteAccepted),
+				Status: metav1.ConditionTrue,
+			}},
+		}}
+		HTTPRouteMustHaveParents(t, c, routeNN, parents, 60)
+	}
+
+	return gwAddr
+}
+
+// WaitForGatewayAddress waits until at least one IP Address has been set in the
+// status of the specified Gateway.
+func WaitForGatewayAddress(t *testing.T, client client.Client, gwName types.NamespacedName, seconds int) (string, error) {
+	t.Helper()
+
+	var ipAddr string
+	waitFor := time.Duration(seconds) * time.Second
+	waitErr := wait.PollImmediate(5*time.Second, waitFor, func() (bool, error) {
+		gw := &v1alpha2.Gateway{}
+		err := client.Get(context.TODO(), gwName, gw)
+		if err != nil {
+			t.Logf("error fetching Gateway: %v", err)
+			return false, fmt.Errorf("error fetching Gateway: %w", err)
+		}
+
+		// TODO: Support more than IPAddress
+		for _, address := range gw.Status.Addresses {
+			if address.Type != nil && *address.Type == v1alpha2.IPAddressType {
+				ipAddr = address.Value
+				return true, nil
+			}
+		}
+
+		return false, nil
+	})
+	require.NoErrorf(t, waitErr, "error waiting for Gateway to have at least one IP address in status")
+	return ipAddr, waitErr
+}
+
+// HTTPRouteMustHaveParents waits for the specified HTTPRoute to have parents
+// in status that match the expected parents. This will cause the test to halt
+// if the specified timeout is exceeded.
+func HTTPRouteMustHaveParents(t *testing.T, client client.Client, routeName types.NamespacedName, parents []v1alpha2.RouteParentStatus, seconds int) {
+	t.Helper()
```

Comment:
Need this within `PollImmediate` to capture the logs from there, if that was the intent

Reply:
Actually that makes it log in wait.go code ... not sure how to get the call of `HTTPRouteMustHaveParents` in the output